src/commands/new.ts

```diff
@@ -620,9 +620,15 @@ module.exports = {
     if (experimentalNewArch === true) {
       startSpinner(" Enabling New Architecture")
       try {
-        let appJsonRaw = read("app.json")
-        appJsonRaw = appJsonRaw.replace(/"newArchEnabled": false/g, '"newArchEnabled": true')
+        const appJsonRaw = read("app.json")
         const appJson = JSON.parse(appJsonRaw)
+        appJson.expo.plugins[1][1].ios.newArchEnabled = true
+        appJson.expo.plugins[1][1].android.newArchEnabled = true
+
+        // Adding the "deploymentTarget" key is required for
+        // @react-native-async-storage/async-storage to work in the new architecture
+        appJson.expo.plugins[1][1].ios.deploymentTarget = "13.4"
```

Comment:
nice regex replace side step!

Reply:
please don't murder me for this array accessor 🫣

Reply:
Could we access it via a find for expo-build-properties instead?

Reply:
@markrickert do you think this helps future proof the array accessor a bit? it's not much cleaner but maybe a little more resilient

```
const expoBuildProperties = appJson.expo.plugins.find(plugin => typeof plugin === 'object' && plugin[0] === 'expo-build-properties')

// access it via expoBuildProperties.ios......
```

Reply:
Well, since we control the generation of this file, I think it's OK to leave as is. I'm actually going to break this out into it's own pull request since it doesn't have anything to do with FlashList, just the new architecture.